### DISCUSSION TITLE
"value has been moved" error in array exercise

### DISCUSSION THREAD

**Discussion by palango** [google/comprehensive-rust#1546] [Q&A] [answered]
Hi,
I was going through the exercises and came across the [`transpose` example](https://google.github.io/comprehensive-rust/tuples-and-arrays/exercise.html):
> The transpose function takes its argument by value, but we haven’t covered ownership yet. Try printing a matrix after it has been transposed, to show the “value has been moved” error, as a preview of ownership and move semantics.

However, I cannot make this error appear, even with using the object multiple times. Any hints?
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0428131bcd42995b27f5fdd43ba2fbdc

**Comment by djmitche** [comment]
Huh, thanks for spotting that! The type of `matrix` is `Copy`, so this error does not appear. I wrote that, so I'm sorry :)

This text was in place to help students understand what that error message would mean, so that it doesn't occur is actually good. I'll take the paragraph out.

**Reply by djmitche** [reply, marked as answer]
#1547.

**Reply by mgeisler** [reply]
Thanks for reporting this!
